services/jupyterhub/spawners.py

```diff
@@ -187,11 +179,12 @@ def start(self):
                 entrypoint='sh -c',
                 command=[
                     'git clone {repository} {volume_path} && '
-                    'git checkout -b {environment_slug} {commit_sha} && '
+                    'git checkout -b {branch}-{username} {commit_sha} && '
```

Comment:
I would use only `{branch}` in this case.

Reply:
then you can't check out a specific commit (branch already exists). We could just checkout the specific commit and not be on a branch at all I suppose.